coreneuron/utils/reports/nrnreport.h

```diff
@@ -37,117 +37,40 @@ THE POSSIBILITY OF SUCH DAMAGE.
-    double start;
-    double stop;
-    double dt;
-    double dt_report;
-    double mindelay;
-    ReportType type;
-    std::string report_filepath;
-
-  public:
-    ReportGenerator(int type,
-                    double start,
-                    double stop,
-                    double dt,
-                    double delay,
-                    double dt_report,
-                    std::string path);
-
-#ifdef ENABLE_REPORTING
-    void register_report();
-#endif
-
-    ~ReportGenerator() {
-        events.clear();
-    }
-};
-
-#if 0
-
-/** type to store every section and associated segments */
-typedef std::vector<int> segment_vector_type;
-typedef std::map<int, segment_vector_type> section_segment_map_type;
-
-/** Mapping information for single neuron */
-class NeuronMappingInfo {
-  public:
-    int gid;           // gid of cellgroup
-    int nsegment;      // no of segments
-    int nsoma;         // no of somas
-    int naxon;         // no of axons
-    int ndendrite;     // no of dendrites
-    int napical;       // no of apical
-    int ncompartment;  // no of compartment
-
-    section_segment_map_type sec_seg_map;  // mapping of section to segments
-
-    NeuronMappingInfo(int id, int seg, int soma, int axon, int dend, int apical, int compartment)
-        : gid(id),
-          nsegment(seg),
-          nsoma(soma),
-          naxon(axon),
-          ndendrite(dend),
-          napical(apical),
-          ncompartment(compartment) {
-    }
-
-    void add_segment(int sec, int seg) {
-        sec_seg_map[sec].push_back(seg);
-    }
-
-    /** section 0 is always soma. there could be multiple compartments
-     *  in the soma and hence return the first compartment
-     */
-    int get_soma_compartment_index() {
-        return (sec_seg_map.begin()->second)[0];
-    }
-};
-
-/** Mapping information for all neurons in NrnThread */
-class NeuronGroupMappingInfo {
-  public:
-    std::vector<NeuronMappingInfo> neuronsmapinfo;  // mapping info for each gid in NrnThread
-
-    void add_neuron_mapping_info(NeuronMappingInfo& mapping) {
-        neuronsmapinfo.push_back(mapping);
-    }
-
-    size_t count() const {
-        return neuronsmapinfo.size();
-    }
-
-    NeuronMappingInfo* get_neuron_mapping(int gid);
-};
-
-#endif
+#include "coreneuron/nrniv/nrn_assert.h"
+#define MAX_REPORT_NAME_LEN 256
+#define MAX_REPORT_PATH_LEN 512
+
+enum ReportType { SomaReport,
+                  CompartmentReport,
+                  SynapseReport
+                };
+
+typedef struct {
+  char            name         [MAX_REPORT_NAME_LEN]; // name of the report
+  char            output_path  [MAX_REPORT_PATH_LEN]; // full path of the report
+  char            target_name  [MAX_REPORT_NAME_LEN]; // target of the report
+  char            mech_name    [MAX_REPORT_NAME_LEN]; // mechanism name
+  char            var_name     [MAX_REPORT_NAME_LEN]; // variable name
+  char            unit         [MAX_REPORT_NAME_LEN]; // unit of the report
```

Comment:
change all this to strings, will be used only in c++ code

Reply:
those are used by the configuration reader